File: ccif/src/java/other/org/himalay/ccif/AcOutlet.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class AcOutlet extends BinStruct implements PublicBinMsg{




    // members variables
    // acOutletControl
    public short acOutletControl ;

        public AcOutlet () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize acOutletControl
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read acOutletControl
        {acOutletControl=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
    
                // write acOutletControl
        ostream.writeByte(acOutletControl); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("AcOutlet\n");
    dc.increaseIndent();
        int retVal= 0;
        // write acOutletControl
        dc.indent();dc.getPs().println("acOutletControl="+acOutletControl+"(0x"+ Integer.toHexString(acOutletControl) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for acOutletControl
    //public short getAcOutletControl()
    //{
    //    return acOutletControl ;
    //}

    
    // Setter for acOutletControl
    //public void setAcOutletControl(short val)
    //{
    //    this.acOutletControl= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/AcOutlet.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "AcOutlet.h"

#ifdef org_himalay_ccif_AcOutlet__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
AcOutlet::AcOutlet() {
    references= 0;


    // acOutletControl
    acOutletControl= (ui8)0;
}
// Destructor
AcOutlet::~AcOutlet() {
    // acOutletControl

}

int AcOutlet::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // acOutletControl
    {
        acOutletControl=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int AcOutlet::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read acOutletControl
    {
        acOutletControl=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int AcOutlet::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write acOutletControl
    ostream.writeUI8(acOutletControl);
    retVal +=1;

    return retVal;
}

int AcOutlet::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("AcOutlet\n");
    dc.increaseIndent();
    int retVal= 0;
    // write acOutletControl
    dc.indent();
    dc.getPs().print("acOutletControl=");
    dc.getPs().println((long)acOutletControl);
    dc.decreaseIndent();
    return retVal;
}


// Getter for acOutletControl
//ui8 AcOutlet::getAcOutletControl()
//{
//return this->acOutletControl;
//}

// Setter for acOutletControl
//void AcOutlet::setAcOutletControl(ui8 val)
//{
//this->acOutletControl= val;
//}

int AcOutlet::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_AcOutlet__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::AcOutlet* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::AcOutlet* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code